src/TLuaInterpreter.cpp

```diff
@@ -5507,112 +5507,135 @@ int TLuaInterpreter::showToolBar(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#sendATCP
 int TLuaInterpreter::sendATCP(lua_State* L)
 {
+    Host& host = getHostFromLua(L);
     string msg;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "sendATCP: what do you want to send?");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "sendATCP: bad argument #1 type (message as string expected, got %1!)", luaL_typename(L,1));
+        return lua_error(L);
     } else {
-        msg = lua_tostring(L, 1);
+        msg = host.mTelnet.encodeAndCookBytes(lua_tostring(L, 1));
     }
 
     string what;
-    if (!lua_isstring(L, 2)) {
-        lua_pushstring(L, "sendATCP: wrong argument type");
-        lua_error(L);
-        return 1;
-    } else {
-        what = lua_tostring(L, 2);
+    if (lua_gettop(L) > 1) {
+        if (!lua_isstring(L, 2)) {
+            lua_pushfstring(L, "sendATCP: bad argument #2 type (what as string is optional, got %1!)", luaL_typename(L,2));
```

Comment:
Well the fact that it is called `what` is the only clue I had to it's purpose - it **is** optional (apparently) so if it is used I have to call it something AND identify it as not being mandatory. The phrasing is, as far as I recall, exactly in accordance with the other (nearly 20) error messages for supplying the wrong argument type for a *non-required* argument...

Reply:
"text to send as string"

Reply:
As I pointed out in another comment below, **`what`** is the name of the argument as described in the wiki and I have no other idea of what `what` means in this usage - someone who understands ATCP can really improve the documentation here I think - and when they have we can change the somewhat ambiguous term used here as well.

> I think we decided against educating the user on what is optional and what's not ...

I am not aware of such a decision (and I am not seeing PRs to change the significant other cases where the message is written exactly like this) - and besides, I reiterate that it does not make sense to me.  If something is optional as an argument then knowing that it can be left out (thus that it is not like other arguments that are required) is as valid/useful a piece of information as knowing what type of argument it should be if it **is** supplied.

Reply:
Ok, I'll raise it separately not to hold this up.